### PR TITLE
Monitor metrics alerts: Support custom namespace. Add guidance for using ANTLR code-gen.

### LINKED ISSUE #9460: 'az vm extension image list --latest' fails
### Version
`azure-cli                         2.0.65`  from the Azure Cloud Shell

### Bug
```
$ az vm extension image list --latest -l westcentralus -o tsv
The command failed with an unexpected error. Here is the traceback:

list index out of range
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 560, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 618, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 611, in _run_job
    six.reraise(*sys.exc_info())
  File "/opt/az/lib/python3.6/site-packages/six.py", line 693, in reraise
    raise value
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 588, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 297, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/__init__.py", line 461, in default_command_handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/vm/custom.py", line 1272, in list_vm_extension_images
    cmd.cli_ctx, publisher_name, name, version, image_location, latest)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/vm/_actions.py", line 155, in load_extension_images_thru_services
    t.result()  # don't use the result but expose exceptions from the threads
  File "/opt/az/lib/python3.6/concurrent/futures/_base.py", line 425, in result
    return self.__get_result()
  File "/opt/az/lib/python3.6/concurrent/futures/_base.py", line 384, in __get_result
    raise self._exception
  File "/opt/az/lib/python3.6/concurrent/futures/thread.py", line 56, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/vm/_actions.py", line 137, in _load_extension_images_from_publisher
    'version': versions[0].name})
IndexError: list index out of range
```

### Details
Using the `az vm extension image list` command with the `--latest` flag generate a traceback, without the flag it returns the expected output.

### PR DESCRIPTION
**Pull request by tjprescott**
Fixes #9460. 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertConditionValidator.py [hunk 1]
```diff
@@ -45,6 +45,7 @@ def exitAggregation(self, ctx):
     # Exit a parse tree produced by MetricAlertConditionParser#namespace.
     def exitNamespace(self, ctx):
         self.parameters['metric_namespace'] = ctx.getText().strip()
+        print(self.parameters['metric_namespace'])
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertConditionValidator.py]: is this `print` expected?

File: src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertConditionListener.py [hunk 2]
```diff
@@ -1,11 +1,4 @@
-# --------------------------------------------------------------------------------------------
-# Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License. See License.txt in the project root for license information.
-# --------------------------------------------------------------------------------------------
-
-# pylint: disable=all
-
-# Generated from MetricAlertCondition.g4 by ANTLR 4.7.1
```
Comment (yugangw-msft) [inline src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertConditionListener.py]: we probably still need the license head

File: src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertConditionValidator.py (hunk 1 above)
Comment (tjprescott) [reply src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertConditionValidator.py]: Ah, no. Thanks!

File: src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertCondition.g4 [hunk 3]
```diff
@@ -8,7 +8,7 @@ expression          : aggregation (namespace '.')* (QUOTE metric QUOTE WHITESPAC
 
 aggregation         : WORD WHITESPACE ;
 
-namespace           : WORD ;
+namespace           : (WORD | '/')+;
```
Comment (AnanyShah) [inline src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertCondition.g4]: @tjprescott Can you please add '.' also

Comment (tjprescott) [reply src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertCondition.g4]: Done

File: src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertConditionValidator.py (hunk 1 above)
Comment (tjprescott) [reply src/azure-cli/azure/cli/command_modules/monitor/grammar/MetricAlertConditionValidator.py]: Fixed.

**Comment by tjprescott** [comment]
@yugangw-msft issues addressed. Should be ready for merge

**Comment by yugangw-msft** [comment]
@tjprescott, can you please resolve the conflict and merge?

**Comment by amitforgit** [comment]
I need to create metric alert in azure CLI:
az monitor metrics alert create
I am not able to pass custom query under --condition.

Can you please suggest if we can use custom condition for az monitor metrics alert create.

Thanks
Amit

**Pull request merged**